lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -345,15 +456,23 @@ def _on_create_backup_action(self, event: ActionEvent) -> None:
             )
 
             logger.info(f"Backup request submitted with backup-id {new_backup_id}")
+            logger.info(f"Backup completed with backup-id {new_backup_id}")
+
+            if wait_for_completion and self.is_backup_in_progress():
+                event.fail(f"Timed out while waiting for: backup id {new_backup_id}")
+                return
         except (
-            RetryError,
-            ValueError,
             OpenSearchHttpError,
-            requests.HTTPError,
             OpenSearchListBackupError,
         ) as e:
             event.fail(f"Failed with exception: {e}")
             return
+        if wait_for_completion:
```

Comment:
this can be put below the previous [condition](https://github.com/canonical/opensearch-operator/pull/158/files#diff-9a3ba6c99d30f70486eef55dba2362fbc0480e6752eb8bb98db1c2e5b24242b4R461)

Reply:
I did not get it. This is here just to check at the end if we should say "backup completed", because we waited for it; or is this an async test and we inform "backup is running"

Reply:
Line 443 you can do:
```
if wait_for_completion:
    if self.is_backup_in_progress():
        event.fail(
            f"Timed out while waiting for: backup id {new_backup_id} - Please check the status again in the future."
        )
    else:
        event.set_results({"backup-id": new_backup_id, "status": "Backup completed."})
else:
    event.set_results({"backup-id": new_backup_id, "status": "Backup is running."})

     
```